firmware/examples/Post/source/main.cpp

```diff
@@ -0,0 +1,287 @@
+
+constexpr uint8_t kAccelerometerAddress = 0x1C;
+constexpr uint8_t kGestureAddress       = 0x39;
+constexpr uint8_t kTemperatureAddress   = 0x31;
+
+I2c i2c;
+
+bool ExternalFlashTest();
+bool I2cTest();
+void OnBoardLedTest(bool gpio_test, bool i2c_test, bool flash_test);
+bool GpioFunctionalityTest();
+bool GestureSensorTest();
+bool AccelerometerTest();
+bool TemperatureSensorTest();
+void OledTest();
+
+int main(void)
+{
+  OledTest();
+  bool gpio_pass  = GpioFunctionalityTest();
+  bool i2c_pass   = I2cTest();
+  bool flash_test = ExternalFlashTest();
+  OnBoardLedTest(gpio_pass, i2c_pass, flash_test);
+
+  // LOG_INFO("Running loop.");
+  Delay(5000);
+  return 0;
+}
+
+void OledTest()
+{
+  Ssd1306 display;
+  LOG_INFO("Starting OLED Hardware Test...");
+
+  LOG_INFO("Initializing OLED Hardware Test...");
+  display.Initialize();
+
+  LOG_INFO("Clearing Screen...");
+  display.Clear();
+
+  LOG_INFO("Clearing Screen finished. Waiting 1 second...");
+
+  Delay(1000);
+  display.Update();
+  LOG_INFO("Inverting screen...");
+  display.InvertScreenColor();
+}
+
+bool ExternalFlashTest()
+{
+  printf("++++++++++++++++++++++++++++++++++++++\n\n");
+  printf("Starting External Flash Test...\n\n");
+  Ssp ssp2(Ssp::Peripheral::kSsp2);
+  Gpio cs(1, 10);
+  cs.SetAsOutput();
+  cs.SetHigh();
+
+  ssp2.SetPeripheralMode(Ssp::MasterSlaveMode::kMaster, Ssp::FrameMode::kSpi,
+                         Ssp::DataSize::kEight);
+  ssp2.SetClock(false, false, 1, 48);
+  ssp2.Initialize();
+  Delay(1000);
+
+  uint8_t array[5];
+  cs.SetLow();
+  Delay(1);
+  // Read Manufacturer ID
+  ssp2.Transfer(0x9F);
+  array[0] = static_cast<uint8_t>(ssp2.Transfer(0x00));
+  array[1] = static_cast<uint8_t>(ssp2.Transfer(0x00));
+  array[2] = static_cast<uint8_t>(ssp2.Transfer(0x00));
+  array[3] = static_cast<uint8_t>(ssp2.Transfer(0x00));
+  LOG_INFO("Returned 0x%02X 0x%02X 0x%02X 0x%02X", array[0], array[1], array[2],
+           array[3]);
+  cs.SetHigh();
+  Delay(1000);
+  LOG_INFO("End of External Flash Test.");
+  return true;
+}
+
+bool I2cTest()
+{
+  // I2C device scan.
+  i2c.Initialize();
+  // Creates a status variable for I2C transactions
+  Status status;
+  printf("++++++++++++++++++++++++++++++++++++++\n\n");
+  printf("Beginning I2c scan...\n\n");
+  for (uint8_t address = kFirstI2cAddress; address < kLastI2cAddress; address++)
+  {
+    status = i2c.Write(address, nullptr, 0);
+    if (status == Status::kSuccess)
+    {
+      LOG_INFO("    Found device at address: 0x%02X", address);
+    }
+  }
+  bool gesture_pass       = GestureSensorTest();
+  bool accelerometer_pass = AccelerometerTest();
+  bool temp_pass          = TemperatureSensorTest();
+  bool final_result       = gesture_pass && accelerometer_pass && temp_pass;
+  return final_result;
+}
+
+bool GpioFunctionalityTest()
+{
+  // P0_2 and P0_3 do not need to be tested.
+  // P0_29 and P0_30 do not need to be tested (lack of pullup/pulldown)
+  const bool LQFP80[5][32] = {
```

Comment:
It might be more readable to use this array designator syntax for the 2D arrays. See: https://en.cppreference.com/w/c/language/array_initialization